### DISCUSSION TITLE
Git Diff is not showing function names that was affected by a change with hunk header for C# file

### DISCUSSION THREAD

**Discussion by mohanpanigrahi** [git-for-windows/git#3722] [Q&A]
Hi,

I am using .gitattributes (*.cs     diff=csharp) for getting name of functions in git diff, it works fine with git version 2.27.0.windows.1. 
When upgraded to git version 2.30.2.windows.1, git diff not giving function name in hunk header. Is there a way to get around this issue?

**git diff --unified=0  $commitID^!  $filePath** or **git diff --function-context $commitID^!  $filePath**

---------------------------- git version 2.27.0.windows.1-----------------------------------
diff --git a/src/BusinessLogic/Extensions/GeneralExtensions.cs b/src/BusinessLogic/Extensions/GeneralExtensions.cs
index c126e01f..637ee3b4 100644
--- a/src/BusinessLogic/Extensions/GeneralExtensions.cs
+++ b/src/BusinessLogic/Extensions/GeneralExtensions.cs
**@@ -44,3 +44,3 @@ public static string ToTitleCaseChange(this string title)**

------------------------------------------git version 2.30.2.windows.1--------------------------------------
diff --cc src/BusinessLogic/Extensions/GeneralExtensions.cs
index 637ee3b4,637ee3b4..c126e01f
--- a/src/BusinessLogic/Extensions/GeneralExtensions.cs
+++ b/src/BusinessLogic/Extensions/GeneralExtensions.cs
**@@@ -44,3 -44,3 +44,3 @@@ namespace Bpp.Applications.Business.C**

**Comment by PhilipOakley** [comment]
The c# settings are in the [git\userdiff.c](https://github.com/git/git/blob/master/userdiff.c#L76-L90) file. I don't believe they have changed recently (RelNotes: in 2.17 recognizes "async" keyword; in 2.33.0 learned the token "record").

Your two comparison use different command lines and file order, which may have an impact.   The `@@@` chunk header is for the (`--cc`) [Combined diff format](https://git-scm.com/docs/git-diff#_combined_diff_format) (section 4.)

**Reply by mohanpanigrahi** [reply]
Thank you Philip for the clue. I used just one (command + parameter) 'git diff --unified=0 $commitID^! $filePath' but getting different output in two different computer having different version of git. ($CommitID and $filepath are same value in both the case).

So I am still not very clear about "different command lines and file order". Would you please elaborate?

Other way, how to prevent use of combined diff format while diffing.  My objective is to get the function names which are changed with hunk headers.

Mohan

**Comment by dscho** [comment]
> `diff --cc src/BusinessLogic/Extensions/GeneralExtensions.cs`

Please note the `--cc`. You are looking at a diff in the presence of merge conflicts. I don't think that function name detection _ever_ worked in the presence of conflicts.

The first diff does not show that option, therefore it is fine.
